_posts/2015-12-01-minutes-offline-case-study.md

```diff
@@ -0,0 +1,101 @@
+
+In addition, caching everything locally (a prerequisite for this entire maneuver) has other positive effects: since the app and its data are present in local browser storage after the first visit, fetching the app, its assets and the data is mostly a zero-latency affair. And even if you’re not Amazon, [losing 1% sales per 100ms lag](http://highscalability.com/latency-everywhere-and-it-costs-you-sales-how-crush-it), you’ll admit that [speed does matter](http://uk.businessinsider.com/facebook-2g-tuesdays-to-slow-employee-internet-speeds-down-2015-10?op=1?r=US&IR=T), because besides faster being better in terms of user experience, network speed and transfer cost are also accessibility issues for many people around the world.
+
+# Offline basics
+
+An offline-first app needs to deal with two separate offlining problems:
+
+  1. Making the application and its assets available offline
+  2. Making each user’s data available offline
+
+We’ll start with the app and its assets. What is minutes.io, and what does it need to do its job?
+
+# minutes.io and its assets
+
+The app itself is a fairly standard single-page application written in Backbone/Marionette, because that’s the only thing available when we started. The app and asset offlining is done with AppCache, and we use Grunt as a build system to, among other things, deal with AppCache’s [many annoyances](http://alistapart.com/article/application-cache-is-a-douchebag).
+
+AppCache requires an up-to-date and complete manifest, and we use [grunt-appcache](https://github.com/canvace/grunt-appcache) to add all the assets to the manifest’s cache section, and all the external assets (like our backend api, Errorception and Intercom) to the network section. We don’t declare a fallback, because the app itself will be fully present and mostly functional anyway.
+
+To make AppCache worry-free in the front-end, we use our own [appcache-nanny](https://github.com/gr2m/appcache-nanny). It does two things:
+
+  - Move the manifest out of the project’s html into an extra [iframe](http://labs.ft.com/2012/11/using-an-iframe-to-stop-app-cache-storing-masters/), so maintaining the manifest is less hassle, and it can shoot itself in the foot less:
```

Comment:
I don't like "shoot itself in the foot", it's quite violent. But I'm happy to hear other peoples opinions on this.